generators/gulp/templates/gulpfile.js

```diff
@@ -12,6 +12,7 @@ var coveralls = require('gulp-coveralls');
 <% } -%>
 <% if (babel) { -%>
 var babel = require('gulp-babel');
+require('babel-core/register');
```

Comment:
Honestly I just moved the original text contained in the mocha config to a require statement at the head of the file

```
<% if (babel) { -%>
    .pipe(mocha({reporter: 'spec', require: ['babel-core/register']}))
    <% } else { -%>
    .pipe(mocha({reporter: 'spec'}))
    <% } %>
```
At this stage there is no compelling reason for the require('babel-core/register') over require('babel/register') as they are both essentially the same.  Happy to make the change or alternatively explicitly use the compiler option when configuring mocha with the babel option which makes the intention very clear as to what is happening e.g.

```
var babelCompiler = require('babel/register');

/* The mocha initialisation within the pipe */ 
mocha({
    compilers: {
        js: babelCompiler
     }
})
```